c29146185.lua

```diff
@@ -27,7 +27,9 @@ end
 function c29146185.retcon(e,tp,eg,ep,ev,re,r,rp)
 	if not re then return false end
 	local rc=re:GetHandler()
-	return rc:IsRace(RACE_SPELLCASTER) or (rc:IsSetCard(0x106e) and rc:IsType(TYPE_SPELL))
+	local loc,race=Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION,CHAININFO_TRIGGERING_RACE)
+	if rc:IsLocation(loc) then race=rc:GetRace() end
```

Comment:
should we check rc is related to re?

Reply:
what does it mean?

Reply:
When monsters on field changed DNA to warrior, _Witchcrafter Edel_ release itself to summon _World of Prophecy_, should _World of Prophecy_ be able to use effect?
If I chain _Call of the Haunted_ and summon _Witchcrafter Edel_ back to field, `rc:IsLocation(loc)` will return true, which race should be used?

Reply:
confirming

Reply:
> Q.
> アンデット族を宣言して発動した「DNA改造手術」が相手フィールドに表側表示で存在する状態で、「魔導獣士 ルード」を対象として自分フィールドのアンデット族となっている「ウィッチクラフト・エーデル」の②の効果を発動しました。
> チェーンしてその効果を発動した「ウィッチクラフト・エーデル」を対象として自分が「戦線復帰」を発動し、さらにチェーンして「DNA改造手術」を墓地へ送って相手が「非常食」を発動しました。
> 「戦線復帰」の効果処理で「ウィッチクラフト・エーデル」が特殊召喚され、「ウィッチクラフト・エーデル」の効果処理のタイミングでその「ウィッチクラフト・エーデル」が魔法使い族としてフィールドに存在していますが、この時「ウィッチクラフト・エーデル」の効果で特殊召喚された「魔導獣士 ルード」の効果を発動できますか？
> A.
> ご質問の場合、「魔導獣士 ルード」の効果は発動できません。